Common/WebStore.Logger/Log4NetLogger.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Reflection;
+using System.Xml;
+using log4net;
+using Microsoft.Extensions.Logging;
+
+namespace WebStore.Logger
+{
+    public class Log4NetLogger : ILogger
+    {
+        private readonly ILog _Log;
+        public Log4NetLogger(string CategoryName, XmlElement Configuration)
+        {
+            var logger_repository = LogManager.CreateRepository(
+                Assembly.GetEntryAssembly(),
+                typeof(log4net.Repository.Hierarchy.Hierarchy));
+
+            _Log = LogManager.GetLogger(logger_repository.Name, CategoryName);
+
+            log4net.Config.XmlConfigurator.Configure(logger_repository, Configuration);
+        }
+
+        public IDisposable BeginScope<TState>(TState State) => null; // логгер не будет поддерживать областей журналов
+
+        public bool IsEnabled(LogLevel Level) // разрешено ли ведение журнала для заданного уровня
+        {
+            switch (Level)
+            {
+                default: throw new ArgumentOutOfRangeException(nameof(Level), Level, null);
+                case LogLevel.None: return false;
+                case LogLevel.Trace: return _Log.IsDebugEnabled;
+                case LogLevel.Debug: return _Log.IsDebugEnabled;
+                case LogLevel.Information: return _Log.IsInfoEnabled;
+                case LogLevel.Warning: return _Log.IsWarnEnabled;
+                case LogLevel.Error: return _Log.IsErrorEnabled;
+                case LogLevel.Critical: return _Log.IsFatalEnabled;
+            }
+        }
```

Comment:
Можно ещё более коротко записать:
```suggestion
        public bool IsEnabled(LogLevel Level) => Level switch
        {
            LogLevel.None => false,
            LogLevel.Trace => _Log.IsDebugEnabled,
            LogLevel.Debug => _Log.IsDebugEnabled,
            LogLevel.Information => _Log.IsInfoEnabled,
            LogLevel.Warning => _Log.IsWarnEnabled,
            LogLevel.Error => _Log.IsErrorEnabled,
            LogLevel.Critical => _Log.IsFatalEnabled,
            _ => throw new ArgumentOutOfRangeException(nameof(Level), Level, null),
        };
```